### PR TITLE
修复tf2运行mlm时的bug

### PR DESCRIPTION
**Pull request by Qinbf**
1.主要是修复了使用tf2运行with_mlm=True程序会产生Inputs to eager execution function cannot be Keras symbolic tensors, but found [<tf.Tensor 'MLM-Proba/transpose:0' shape=(768, 21128) dtype=float32>]，类似的由于tf2的动态图机制导致的错误。我只测试了bert模型。其他模型的EmbeddingDense我也做了类似修改。
2.另外改了一下bert模型with_nsp=True，如果没有设置with_pool=True会导致activations.get报错。

### CODE REVIEW

**Comment by bojone** [comment]
你能不能先给出一个会出错的简单demo以及具体的软件环境？我在TF 2.0下测了basic_masked_language_model.py是不会报错的。

你提的修改方式是不适合的，因为最后的Dense层要与Embedding层共享kernel，而你的方案没办法实现这一点。

**Comment by Qinbf** [comment]
mac系统，TF2.1.0环境，python3.6下面这段程序就会报错。

`import os
os.environ['TF_KERAS'] = '1'
from bert4keras.models import build_transformer_model
from bert4keras.tokenizers import Tokenizer
import numpy as np
from plot_model import plot_model

model_dir = './chinese_roberta_wwm_ext_L-12_H-768_A-12'

config_path = model_dir+'/bert_config.json'
checkpoint_path = model_dir+'/bert_model.ckpt'
dict_path = model_dir+'/vocab.txt'

tokenizer = Tokenizer(dict_path) # 建立分词器
model = build_transformer_model(config_path, checkpoint_path, with_mlm=True) # 建立模型，加载权重

token_ids, segment_ids = tokenizer.encode('机器学习是一门计算机科学')

token_ids[4] = token_ids[5] = tokenizer._token_dict['[MASK]']

# 用mlm模型预测被mask掉的部分
probas = model.predict([np.array([token_ids]), np.array([segment_ids])])[0]
print(tokenizer.decode(probas[4:6].argmax(axis=1)))  `

确实我的修改只考虑了模型载入部分，训练阶段是分别训练的，改的时候忘记考虑这点了。不过理论上这两部分是可以分开训练的，并且分开训练说不定效果更好（这是我胡说的，没有做过实验哈哈）。你可以先看看这个问题，具体怎么改你再考虑考虑。

**Comment by bojone** [comment]
我手头上也有py3+tf2.0的docker，刚测试了没有问题。所以看来问题是在tf 2.1时才出现的，我再测试一下。

**Comment by Qinbf** [comment]
确实，我刚测了tf2.0，没毛病。tf2.1不行

**Comment by bojone** [comment]
还是感谢你的反馈，这个pull先关了，后面我在tf 2.1测试后再想办法修复。

**Comment by Qinbf** [comment]
好
